app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :post_owner?, only: %i[edit update destroy]
+
+  def index
+    @post = Post.new
+    @posts = Post.all
+  end
+
+  def show; end
+
+  def new
+    @post = Post.new
+  end
+
+  def edit; end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      # flash[:notice] = 'Post was successfully created.'
```

Comment:
It would be cool to see you flash notification to the user upon the creation of a post. It's `not required` to pass on this milestone though. Pending when you want to implement it, you should leave your code free of commented code like this.